.github/workflows/build-wrapper.yaml

```diff
@@ -0,0 +1,33 @@
+name: "[post-commit] Build C++ binaries with all configs"
+
+on:
+  workflow_call:
+  workflow_dispatch:
+  push:
+    branches: ["mchiou/11762*"]
+
+permissions:
+  actions: read
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
Mmmm do we still need `id-token: write` here? If so what for?

Reply:
slack notifications if this is run on main branch.
there's some token that the action will need in order to pull slack api token to push to our slack channels